### PR TITLE
agent: allow changing file ownership in file sink

### PR DESCRIPTION
**Pull request by clwluvw**
Allow changing the ownership of the token file in file sink.

### CODE REVIEW

File: website/content/docs/agent-and-proxy/autoauth/sinks/file.mdx [hunk 1]
```diff
@@ -21,7 +21,9 @@ written with `0640` permissions as default, but can be overridden with the optio
 ## Configuration
 
 - `path` `(string: required)` - The path to use to write the token file
-- `mode` `(int: optional)` - A string containing an octal number representing the bit pattern for the file mode, similar to chmod. Set to `0000` to prevent Vault from modifying the file mode. Note: This configuration option is only available in Vault 1.3.0 and above.
+- `mode` `(int: optional)` - A string containing an octal number representing the bit pattern for the file mode, similar to chmod. Note: This configuration option is only available in Vault 1.3.0 and above.
```
Comment (schavis) [inline website/content/docs/agent-and-proxy/autoauth/sinks/file.mdx:24]:
```suggestion
- `mode` `(int: optional)` - Octal number string representing the bit pattern for the file mode, similar to `chmod`.
```

The Vault docs only go back to v1.4, so I'm not sure the callout is necessary

**Review by schavis — CHANGES_REQUESTED** [review]

File: website/content/docs/agent-and-proxy/autoauth/sinks/file.mdx [hunk 2]
```diff
@@ -21,7 +21,9 @@ written with `0640` permissions as default, but can be overridden with the optio
 ## Configuration
 
 - `path` `(string: required)` - The path to use to write the token file
-- `mode` `(int: optional)` - A string containing an octal number representing the bit pattern for the file mode, similar to chmod. Set to `0000` to prevent Vault from modifying the file mode. Note: This configuration option is only available in Vault 1.3.0 and above.
+- `mode` `(int: optional)` - Octal number string representing the bit pattern for the file mode, similar to `chmod`.
+- `owner` `(int: optional)` - The UID to use for the token file. Defaults to the current user ID.
+- `group` `(int: optional)` - The GID to use for token file. Defaults to the current group id.
```
Comment (VioletHynes) [inline website/content/docs/agent-and-proxy/autoauth/sinks/file.mdx:26]:
```suggestion
- `group` `(int: optional)` - The GID to use for token file. Defaults to the current group ID.
```

**Review by VioletHynes — APPROVED** [review]
Hey there! This looks great, thank you for this.

I'm going to add a changelog and try and get this merged shortly.

File: changelog/27123.txt [hunk 3]
```diff
@@ -0,0 +1,7 @@
+```release-note:improvement
+agent/sink: Allow configuration of the user and group ID of the file sink.
+```
+```release-note:improvement
+proxy/sink: Allow configuration of the user and group ID of the file sink.
+```
+
```
Comment (VioletHynes) [inline changelog/27123.txt:7]:
```suggestion
```
```

Comment (VioletHynes) [inline changelog/27123.txt:7]:
```suggestion
```
```

**Comment by VioletHynes** [comment]
Hi there @clwluvw ! Just one little thing before we get this merged:

the linter check failed with the following:
`command/agentproxyshared/sink/file/file_sink_test.go:139:1: Test TestFileSinkMode_Ownership is missing a go doc`

Could you make sure to add a godoc for the test, describing what it does and what it's for? Thanks :D

I also need to run this by @schavis to make sure there are no outstanding requested changes.

**Comment by clwluvw** [comment]
Hi @VioletHynes - Thanks for the review. I just added the godoc for the test.

**Comment by VioletHynes** [comment]
Thanks! It's a holiday in some parts of the world today and I'll need to confirm with Sarah before I merge this, but I hope to get this merged soon this week :)

**Review by schavis — APPROVED** [review]

**Pull request merged**